runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcGetDataStream.java

```diff
@@ -171,7 +182,9 @@ public GlobalData requestGlobalData(GlobalDataRequest request) {
   }
 
   @Override
-  public void refreshActiveWork(Map<String, List<KeyedGetDataRequest>> active) {
+  public void refreshActiveWork(
+      Map<String, List<KeyedGetDataRequest>> active,
+      Map<String, List<HeartbeatRequest>> heartbeats) {
     long builderBytes = 0;
```

Comment:
Request will get rejected if both KeyedGetDataRequest and Heartbeat request are present at the same time. If its easy, can we assert that only one map is not empty here?

Reply:
Code has been refactored to only have a single Map now, so I guess this is obsolete.